gui/src/app/CompilationServerConnectionControl/ConfigureCompilationServerDialog.tsx

```diff
@@ -1,38 +1,36 @@
-import { FunctionComponent, useCallback } from "react";
-import {
-  localUrl,
-  publicUrl,
-  ServerType,
-} from "./CompilationServerConnectionControl";
-import FormControl from "@mui/material/FormControl";
+import { CompileContext } from "@SpCompileContext/CompileContext";
+import { Refresh } from "@mui/icons-material";
 import Divider from "@mui/material/Divider";
-import FormLabel from "@mui/material/FormLabel";
-import RadioGroup from "@mui/material/RadioGroup";
+import FormControl from "@mui/material/FormControl";
 import FormControlLabel from "@mui/material/FormControlLabel";
+import FormLabel from "@mui/material/FormLabel";
+import IconButton from "@mui/material/IconButton";
 import Radio from "@mui/material/Radio";
+import RadioGroup from "@mui/material/RadioGroup";
 import TextField from "@mui/material/TextField";
-import IconButton from "@mui/material/IconButton";
-import { Refresh } from "@mui/icons-material";
+import { FunctionComponent, useCallback, useContext } from "react";
+import { localUrl, publicUrl } from "./CompilationServerConnectionControl";
+
+type ServerType = "public" | "local" | "custom";
 
 type ConfigureCompilationServerDialogProps = {
-  stanWasmServerUrl: string;
-  setStanWasmServerUrl: (url: string) => void;
   isConnected: boolean;
   onRetry: () => void;
-  choice: ServerType;
-  setChoice: (choice: ServerType) => void;
 };
 
 const ConfigureCompilationServerDialog: FunctionComponent<
   ConfigureCompilationServerDialogProps
-> = ({
-  stanWasmServerUrl,
-  setStanWasmServerUrl,
-  isConnected,
-  onRetry,
-  choice,
-  setChoice,
-}) => {
+> = ({ isConnected, onRetry }) => {
+  const { stanWasmServerUrl, setStanWasmServerUrl } =
+    useContext(CompileContext);
+
+  const serverType: ServerType =
+    stanWasmServerUrl === publicUrl
+      ? "public"
+      : stanWasmServerUrl === localUrl
+        ? "local"
+        : "custom";
```

Comment:
We repeat this snippet of code in two places, might be worth pulling into a function

Reply:
Done